### ISSUE TITLE
Taking the user by the hand

### ISSUE THREAD

**Issue by jacogr** [polkadot-js/apps#900]
Currently not doing a great job for getting new users on-board and explaining what is happening where. (This probably will explode into separate issues since there will be multiples originating from this, but starting here)

- First use: Here we have no accounts, so certain actions are not visible - however it is not obvious that the user needs to have accounts. For first use, you probably want an on-board dialog that actually helps the user "unlock stuff" by creating an account. (And probably just give some board strokes info of where to get more info)

- Apps: (very broad) For some stuff, we have "label help" now, e.g. staking. The suggestion is to expand this throughout, just to give me hints on what is what. (Focus being on democracy, staking, transfer - state, toolbox & extrinsics assume a higher-level, so can be left initially - and generally dynamic anyway)

- ... there are more, help needed to expand...

Not of this is cast in stone, just doing a brain dump - whatever works best UX-wise.
 
@rossbulat @kwingram25  Would appreciate your insights and additions as well for new users.

**Comment by rossbulat** [comment]
I am seeing an additional section as a landing page, perhaps called "Home". "Home" is a very familiar term to users, and has less ambiguity than other terms like "Dashboard".

A Home app would abstract the other apps and curate them in a way that the user can follow. Along with help prompts and status settings, content-wise could include in the current state:

* Transfer button to open full-screen transfer app
    * no need for SideBar - have back or cross icon to default back home
    * opportunity to remove Transfer from sidebar completely, simplifying navigation
* A list of the users accounts on this page, of which clicking an account goes to the Accounts app, with Create First Account button in the event no accounts are imported.
* Validator status and link to validator app.
* Active proposals and referendums that go to the democracy app.

So mostly curating the services of the app so far - displaying possible actions rather than adding additional functionality at this stage.

**Comment by rossbulat** [comment]
Glossary will be very handy for newcomers.
Possible implementation: 
* React Suspense based glossary that loads asynchronously as users request to view, whereby no additional loading time is added to initial page load.
* Perhaps an overlay that slides from the right side of page (akin to Amazon help), that does not disturb where the user currently is in the app.
* filtering of glossary terms
* Adopt markup for glossary file for easy management / contributions on Github
* Keep "Web3'y" - host glossary inside the app, as opposed to an API to a central server.

**Comment by jacogr** [comment]
Thanks @rossbulat 

The glossary/help idea on a per app basic is actually great. It can have all the info required to explain wtf it is, how it works.

I'm not 100% convinced by the "Home" - the first reason is that we did something like that for the now-defunct Ethereum wallet and really, not mnay spend a lot of time there. The navigation should just be simple enough.

The secondary reason is that apps really adjusts on a per-chain basis, showing what is available, nothing is or should be static apart for the fact that something like accounts exist - which is actually also something that may not even be there.  So what you described works for Polkadot, but not in the cases on for instance utxo chains where transfer doesn't exist (`balances.*` module is not there), or for chains where there are no balances (something that can be supported). At the same time democracy is also fluent, it may be there, may be removed completely (no `democracy.*`) or replaced.

So the idea of "Home" could end up being completely empty. It is the beauty of substrate and what makes things a bit difficult - what you see on screen may or may not exist. (Depending on chain, config and adjustments.)

**Comment by rossbulat** [comment]
> The glossary/help idea on a per app basic is actually great. It can have all the info required to explain wtf it is, how it works.

Indeed, I guess it will be viable to link each term to external sources for deeper explanations too.

> I'm not 100% convinced by the "Home" - the first reason is that we did something like that for the now-defunct Ethereum wallet and really, not many spend a lot of time there. The navigation should just be simple enough.

> The secondary reason is that apps really adjusts on a per-chain basis,

Understood, I guess the SideBar itself can be seen as a Home section; it currently displays and hides sections depending on apps available. I can picture an expanded bar with more details similar to what is in #376.

**Comment by jacogr** [comment]
Yes, make no mistake, don't think the SideBar is perfect as a navigation aid atm, but it does really play that role of making available what is on-chain.

**Comment by chevdor** [comment]
I have been already thinking about that but always ended up with the conclusion that the UI is OK and improvements will always with biased depending on who the user is. Your post @jacogr made me rethink.

First I will assume we are talking about the Polkadot UI and not a generic Substrate chain UI.
Initially, all the functions have been implemented in a generic way so that everything works but this was not really user friendly. There has been amazing improvements since and the current version.

IMO, we should distinguish WHO is using the UI and I think this could be split as follow.

First you mention the need to take the user by hand. In that case, we talk about the beginner user (no offence!). Making a UI optimised towards beginners is usually great for new users but sucks once users get around the project.

I would propose to make a cut between *users* (= standard user and beginners) and *advanced users*. Unless the user is an advanced one, I would see a great benefit in hiding things like the toolbox, javascript and chainstate tabs.

I like the idea of a home/dashboard. Today, the UI is split as the software is which may not bring the best UX. 

I think a dashboard would be beneficial to the users if implemented properly. If we consider the democracy functions for instance. A regular user will never need to see anything around that *except* if there is a referendum im progress. That's where a dashboard can be useful by showing something but only when a referendum is in progress.

The role of such dashboard should be to aggregate the infos we currently have in many tabs.

I think the UI can also be made more user friendly once we know what profile has the user. We can surely make a cut between people who will aim at being a validator and the others which are token holders and nominators. 

Tabs such as the chain state are more of an *advanced user* or even *dev* function. Standard users will not go look in there but expect a widget to fetch meaning data for them.

I am not really sure how complex the task would be but I would find awesome to have a dashboard similar to how grafana works and the ability for the users to build it themselves.

Mine for instance, would be make of a bit of the current explorer tab, a bit of staking, likely a few from chain state and a bit part reserved to the accounts, balances and staking.

Once we have a home/dashboard which is the rally location for all users, it would be simple to have some kinda alert box if no account is found in local storage and point the user to the appropriate tab as a start. I also think that the beginner user would benefit from direct links to the doc (=wiki).

For instance, as beginner, I would like to see a little sentence explaining what democracy is (ie what can I do/see in this doc and a link to the wiki to know more).

One important point to clarify is likely to define whether the UI is intended to grow with apps (ie each app gets a tab somehow) or with widgets (=grafana panels) that can be organized within more statis apps. 

An easy way to smoothen onbaording can also be something like https://react-joyride.com/controlled. I think this is however rather time consuming and I am not convince the time investment is worth the time the users spend reading versus the time spent to have a wiki (or alike) doc.

**Comment by jacogr** [comment]
@chevdor No offense taken. (This was actually spawned by a convo I had with @kwingram25 - "how do I copy an address"... and he is a nowhere close to a mom-and-pop user)

So starting at the bottom - I'm not quite open for joyride/controlled - just have done stuff like that is the past, and it is great, but really to keep it updated as you move along and tweak and move fast - just breaks way too much. If things were stable, maybe.

So have you tried running the interface in "light" mode? (Settings) There are 2 types of users I want to support -

- the newbies - these grow into the pros, so want to make them comfortable from the get-go
- the pros - lots of devs, lots of people who have been around the block and want to dig and explore and squeeze - and on all types, so I'm still balking at the idea of a "home" / "dashboard" that can be different depending on the chain. (However, code talks, bullshit walks - happy to be convinced that I am wrong, I amd maybe tainted by previous efforts here in the Ethereum world as well)

My main take-aways from your post (and there will be more, just my initial gut reaction) is the following -

- Probably need to really think about making the light (simple mode) the default. And make that "good enough" (We need to get to "good" before we can move to "great". Probably first expand, clean up, make simple - then default)
- Really need to put some honest effort into the guides, e.g. look at staking, click on the (?) top-right. It is a start of helping and guiding people along. (A start... not more. This needs expansion). Same with the help hints all along
- ... more to come

**Comment by chevdor** [comment]
> So starting at the bottom - I'm not quite open for joyride/controlled - just have done stuff like that is the past, and it is great, but really to keep it updated as you move along and tweak and move fast - just breaks way too much. If things were stable, maybe.

I agree

> So have you tried running the interface in "light" mode? (Settings) There are 2 types of users I want to support -

Oh yeah, I forgot that! Maybe the solution is just to make the light mode the default as you mentioned and make it easier to switch.

> the newbies - these grow into the pros, so want to make them comfortable from the get-go
> the pros - lots of devs, lots of people who have been around the block and want to dig and explore and squeeze - and on all types, so I'm still balking at the idea of a "home" / "dashboard" that can be different depending on the chain. (However, code talks, bullshit walks - happy to be convinced that I am wrong, I amd maybe tainted by previous efforts here in the Ethereum world as well)

I would be in favor of a Home but I would indeed not do the Parity UI Home again.

> My main take-aways from your post (and there will be more, just my initial gut reaction) is the following -
> 
> Probably need to really think about making the light (simple mode) the default. And make that "good enough" (We need to get to "good" before we can move to "great". Probably first expand, clean up, make simple - then default)

100%

> Really need to put some honest effort into the guides, e.g. look at staking, click on the (?) top-right. It is a start of helping and guiding people along. (A start... not more. This needs expansion). Same with the help hints all along

I see. The question is if you want to carry the doc inside the UI or simply link to some Wiki.
From what I see on the Staking help, I also think users need smaller chunks of doc to digest it better.

Linking here what NXT is doing, I find their on-boarding simple but it does the trick: https://demo.nxt.org/index.html

**Comment by jacogr** [comment]
The issue with stuff externally is that the same questions keep on popping up. People are lazy. With the in-app help, it needs to have an overview of the various functions (more-or-less what staking has - not as comprehensive as a wiki, maybe with a "learn more" additional link to a wiki), but also need to try and keep it specific and toned towards using the app. (There is no real example of this as of yet.)

And yes, the oboarding is one this that is really missing - first launch, just wtf do I do now? Will take a peek, thanks.

**Comment by chevdor** [comment]
Would it not be an option to:
- have the doc external (good for the devs)
- pull it inside the app (good for users)

**Comment by jacogr** [comment]
Yes, that is basically what I'm suggestion - enough to give you a warm and fuzzy inside the app, with the links to external, dedicated and comprehensive resources.
